services/sonar/sonar-fortify-rating.service.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const SonarBase = require('./sonar-base')
+const {
+  patternBase,
+  queryParamSchema,
+  keywords,
+  documentation,
+} = require('./sonar-helpers')
+
+const colorMap = {
+  0: 'red',
+  1: 'orange',
+  2: 'yellow',
+  3: 'yellowgreen',
+  4: 'green',
+  5: 'brightgreen',
+}
+
+module.exports = class SonarFortifyRating extends SonarBase {
+  static get category() {
+    return 'analysis'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'fortify-security-rating' }
+  }
+
+  static render({ rating }) {
+    return {
+      message: `${rating}/5`,
+      color: colorMap[rating],
+    }
+  }
+
+  static get route() {
+    return {
+      base: 'sonar',
+      pattern: `${patternBase}/fortify-security-rating`,
+      queryParamSchema,
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Sonar Fortify Security Rating',
+        namedParams: {
+          protocol: 'http',
+          host: 'sonar.petalslink.com',
+          component: 'org.ow2.petals:petals-se-ase',
```

Comment:
this  `namedParams` object doesn't seem to generate a valid badge:

https://shields-staging-pr-3189.herokuapp.com/sonar/http/sonar.petalslink.com/org.ow2.petals:petals-se-ase/fortify-security-rating.svg

Reply:
Yeah, we don't know of any live Sonar endpoints that have the Fortify plugin installed as noted above in the PR summary.

The SaaS service sonarcloud.io doesn't have nor does http://sonar.petalslink.com. We don't have a live endpoint that we can use for the example, nor for the tests (that's why they're all mocked)

Reply:
Aah yes - you did mention this elsewhere in the PR :+1:

Reply:
Should I remove the example listing? I'm a fan of the editor experience it provides, but my instinct says that there's not a ton of folks using this badge today.

Alternatively, I could update the documentation. It currently calls out that it def. won't work on SonarCloud, but I could update that to note that the Fortify stuff comes from a Sonar plugin that's not available OOTB

Reply:
I think its ok. If we ever find a public instance with the plugin, it would be nice to update it.
We can review if we find there are hoards of confused Sonar users opening issues.